### PR TITLE
Implement some features and improvments

### PR DESCRIPTION
**Pull request by IvanGael**
** NodeJS Backend ** 
- Configuration management with a config.json file.
- Improved error handling and logging using Winston.
- Screen boundary checks to prevent the cursor from moving off-screen.
- Support for keyboard shortcuts for common presentation actions.
- Multi-client support with a basic room system.
- Laser pointer functionality (broadcast to clients in the same room).

** Flutter app **
- Improve the ui and the code 
- Update the frontend accordingly to the NodeJS backend

### CODE REVIEW

**Comment by cl0v** [comment]
Thanks for your submission, you have removed the notes about the Apache License and this should not be allowed.

**Comment by IvanGael** [comment]
@cl0v  I thought it was just comments forgotten in the code. I wanted to make the code clean and more readable.

Since you say it now I'll put it back in the code.

**Comment by IvanGael** [comment]
The notes about the Apache License is now inlcuded

**Comment by IvanGael** [comment]
Media Control feature... Does it for handling presentations media ?

File: nodejs/src/robot.ts [hunk 1]
```diff
@@ -34,21 +39,15 @@ export const keyboardType = (value: string) => {
 };
 
 export function moveCursor(x: number, y: number) {
+    const { x: newX, y: newY } = checkScreenBoundaries(x, y);
     mousePos = getMousePos();
-
-    console.log(`Mouse Motion (${x},${y})`)
-
-    posx += (screenSize.width / 32) * x
-    posy += (screenSize.height / 18) * y
```
Comment (cl0v) [inline nodejs/src/robot.ts:42]: Those maths are necessary so the movement improves a significant ammount, please put them in the moveCursor back.

**Comment by cl0v** [comment]
Yes, the handling presentations media should be a separate feature.

File: nodejs/src/robot.ts [hunk 2]
```diff
@@ -58,3 +57,31 @@ export function rightClick() {
...
+    }
+}
+
+function checkScreenBoundaries(x: number, y: number): { x: number; y: number } {
+    const currentPos = getMousePos();
+    
+    let newX = Math.max(0, Math.min(screenSize.width, currentPos.x + x));
+    let newY = Math.max(0, Math.min(screenSize.height, currentPos.y + y));
+    
+    return { x: newX - currentPos.x, y: newY - currentPos.y };
```
Comment (cl0v) [inline nodejs/src/robot.ts:86]: Maybe you should try put the maths here

Comment (cl0v) [reply nodejs/src/robot.ts:86]: Like this:

`   
let newX = Math.max(0, Math.min(screenSize.width, currentPos.x + x * 32));
let newY = Math.max(0, Math.min(screenSize.height, currentPos.y + y * 18));
`

**Comment by cl0v** [comment]
I will approve the changes without the need to create a new pull request for the Presentation Medias, but please fix the mouse movement with the multiplies to improve the performance.

**Comment by cl0v** [comment]
I'm not sure about the multi client connection, but I will allow it for now.

**Review by cl0v — APPROVED** [review]
Great job!

**Pull request merged**